addons/payment/utils.py

```diff
@@ -22,9 +24,15 @@ def generate_access_token(*values):
     :return: The generated access token
     :rtype: str
     """
-    token_str = '|'.join(str(val) for val in values)
-    access_token = hmac_tool(request.env(su=True), 'generate_access_token', token_str)
-    return access_token
+    try:
+        request_env = env(su=True) if env else request.env(su=True)
```

Comment:
I am not sure I would use request_env as a variable name because we know that there is not always a http request.